test/http_test.c

```diff
@@ -416,6 +416,63 @@ static int test_http_keep_alive_1_require_no(void)
     return test_http_keep_alive('1', 2, 0);
 }
 
+static int test_http_resp_hdr_limit(unsigned int limit)
+{
+    BIO *wbio = BIO_new(BIO_s_mem());
+    BIO *rbio = BIO_new(BIO_s_mem());
+    BIO *mem = NULL;
+    server_args mock_args = { NULL, NULL, NULL, '0', 0 };
+    int res = 0;
+    OSSL_HTTP_REQ_CTX *rctx = NULL;
+
+    if (wbio == NULL || rbio == NULL)
+        goto err;
+
+    mock_args.txt = text1;
+    mock_args.content_type = "text/plain";
+    mock_args.version = '1';
+    mock_args.out = rbio;
+    mock_args.content_type = "text/plain";
+
+    BIO_set_callback_ex(wbio, http_bio_cb_ex);
+    BIO_set_callback_arg(wbio, (char *)&mock_args);
+
+    rctx = OSSL_HTTP_REQ_CTX_new(wbio, rbio, 8192);
+    if (rctx == NULL)
+        goto err;
+
+    if (!OSSL_HTTP_REQ_CTX_set_request_line(rctx, 0 /* GET */,
+                                            NULL, NULL, RPATH))
+        goto err;
+
+    OSSL_HTTP_REQ_CTX_set_max_response_hdr_lines(rctx, limit);
+    mem = OSSL_HTTP_REQ_CTX_exchange(rctx);
```

Comment:
Where does this mem bio get freed?

Reply:
If I understand `OSSL_HTTP_REQ_CTX(3)` manpage correct, then I'm not supposed to free `mem` here.  Quoting from `RETURN VALUES` section of the manpage:

```
       OSSL_HTTP_REQ_CTX_exchange() and OSSL_HTTP_REQ_CTX_get0_mem_bio()
       return a pointer to a BIO on success as described above or NULL on
       failure.  The returned BIO must not be freed by the caller.
```